pilot/pkg/status/state.go

```diff
@@ -222,11 +221,14 @@ func (c *DistributionController) removeStaleReporters(staleReporters []string) {
 }
 
 func GetTypedStatus(in interface{}) (out v1alpha1.IstioStatus, err error) {
-	var statusBytes []byte
-	if statusBytes, err = json.Marshal(in); err == nil {
-		err = json.Unmarshal(statusBytes, &out)
+	if in == nil {
+		return v1alpha1.IstioStatus{}, fmt.Errorf("status was nil")
 	}
-	return
+	out, ok := in.(v1alpha1.IstioStatus)
```

Comment:
I think `in` is a `*v1alpha1.IstioStatus` (pointer). probably why the test is failing?

Reply:
Looking at `state_test.go`, I see `statusStillPropogating` as a type `v1alpha1.IstioStatus`, not of type `*v1alpha1.IstioStatus` : https://github.com/istio/istio/blob/master/pilot/pkg/status/state_test.go#L25.

`statusStillPropogating` is passed into `GetTypedStatus` in https://github.com/istio/istio/blob/master/pilot/pkg/status/state_test.go#L188, is it really a pointer?

When I change the type assertion to a pointer, a bunch of unit tests fail as well.